src/pages/MainPage/components/DocumentEditor.js

```diff
@@ -0,0 +1,117 @@
+import { push } from "../../../services/router.js";
+
+export default function DocumentEditor({ $target, onEditing, isDocument }) {
+  const $editorWrapper = document.createElement("div");
+  $editorWrapper.className = "docEditWrapper";
+
+  const $title = document.createElement("div");
+  $title.className = "docEditTitle";
+  $title.contentEditable = true;
+
+  const $content = document.createElement("div");
+  $content.className = "docEditContent";
+  $content.contentEditable = true;
+
+  $editorWrapper.append($title, $content);
+  $target.appendChild($editorWrapper);
+
+  this.state = {};
+
+  this.setState = (nextState) => {
+    if (nextState.content === null) nextState.content = "<div><br></div>";
+    this.state = nextState;
+    this.render();
+  };
+
+  this.render = () => {
+    $title.textContent = this.state.title;
+    $content.innerHTML = this.state.content;
+  };
+
+  $title.addEventListener("keydown", (event) => {
+    if (event.code === "Enter") {
+      event.preventDefault();
+      $content.focus();
+    }
+  });
+
+  $title.addEventListener("input", (event) => {
+    this.setState({
+      ...this.state,
+      title: event.target.textContent,
+    });
+
+    onEditing(this.state);
+  });
+
+  $content.addEventListener("input", (event) => {
+    onEditing({
+      ...this.state,
+      content: event.target.innerHTML,
+    });
+  });
+
+  $content.addEventListener("keyup", (event) => {
+    if (event.code === "Backspace" && event.target.textContent === "") {
+      event.target.innerHTML = "<div><br></div>";
+    }
+
+    if (event.code === "Enter") {
+      // if (event.target.textContent.charCodeAt(event.target.textContent.length - 1) >= 12593) {
+      //   $content.dispatchEvent(new KeyboardEvent('keydown', {key: 'KeyA'}));
+      //   console.log('kkk');
+      // }
+      event.preventDefault();
+      const $node = $content;
+      const caretID = "_caret";
+      const $cc = document.createElement("span");
+      $cc.id = caretID;
+      window.getSelection().getRangeAt(0).insertNode($cc);
+      $node.blur();
+
+      const divList = event.target.querySelectorAll("div");
+      [...divList].forEach((div) => {
+        if (div.textContent.indexOf("# ") === 0) {
+          div.innerHTML = `<h1>${div.textContent.substring(2)}</h1>`;
```

Comment:
p3
정규표현식으로 처리 가능할 것 같아요
```suggestion
          div.innerHTML = `<h1>${div.textContent.replace(/^# /g, '')}</h1>`;
```